File: Groove.SP/Groove.SP.Core/Entities/RoleModel.cs
using Groove.SP.Core.Models;
using System.Collections.Generic;

namespace Groove.SP.Core.Entities
{
    public class RoleModel : Entity
    {
        public long Id { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public bool Activated { get; set; }
        public bool IsInternal { get; set; }
        public RoleStatus Status { get; set; }
        public bool IsOfficial { get; set; }

        public virtual ICollection<UserRoleModel> UserRoles { get; set; }
        public virtual ICollection<RolePermissionModel> RolePermissions { get; set; }
        public virtual ICollection<ViewRoleSettingModel> ViewRoleSettings { get; set; }
    }
}

File: Groove.SP/groove-sp-web/src/app/core/models/role/role.model.ts
import { RoleStatus } from '../enums/enums';
import { ModelBase } from '../model-base.model';

export class RoleModel extends ModelBase {
    id: number;
    name: string;
    description: string;
    activated: boolean;
    isInternal: boolean;
    status: RoleStatus;
    isOfficial: boolean;
    statusName: string;
    permissionIds: number[];
}